File: db4on/core/src/core/com/db4o/config/Compare.cs
namespace com.db4o.config
{
	/// <summary>allows special comparison behaviour during query evaluation.</summary>
	/// <remarks>
	/// allows special comparison behaviour during query evaluation.
	/// <br /><br />db4o will use the Object returned by the <code>compare()</code>
	/// method for all query comparisons.
	/// </remarks>
	public interface Compare
	{
		/// <summary>return the Object to be compared during query evaluation.</summary>
		/// <remarks>return the Object to be compared during query evaluation.</remarks>
		object compare();
	}
}

File: db4oj/core/src/com/db4o/config/Compare.java
/* Copyright (C) 2004   Versant Inc.   http://www.db4o.com */

package com.db4o.config;

/**
 * allows special comparison behaviour during query evaluation.
 * <br><br>db4o will use the Object returned by the <code>{@link Compare#compare()}</code>
 * method for all query comparisons.
 */
public interface Compare {
	
	/**
	 * return the Object to be compared during query evaluation.
	 */
	public Object compare();
	
}

